sklearn/ensemble/_stacking.py

```diff
@@ -592,9 +602,21 @@ def fit(self, X, y, sample_weight=None):
             Returns a fitted instance of estimator.
         """
         check_classification_targets(y)
-        self._le = LabelEncoder().fit(y)
-        self.classes_ = self._le.classes_
-        return super().fit(X, self._le.transform(y), sample_weight)
+        self._type_of_target = type_of_target(y)
+        self.classes_ = []
+        if self._type_of_target != "multilabel-indicator":
+            self._le = LabelEncoder().fit(y)
+            self.classes_ = self._le.classes_
```

Comment:
It is worth being verbose for naming private attributes, here `LabelEncoders`.
```suggestion
            self._label_encoder = LabelEncoder().fit(y)
            self.classes_ = self._label_encoder.classes_
```

This suggestion applies in other places.